### ISSUE TITLE
Sweep_: format wandb run name in pytorch checkpoint

### ISSUE THREAD

**Issue by ayulockin** [ayulockin/pytorch-lightning#4]
### Description & Motivation

It would be great to format the wandb run name: 
`wandb.run.name`

into the checkpoint filename.

epoch: 10
loss: 0.5423

filename: "epoch{epoch}-loss{val_loss:.2f}" --> "epoch10-loss0.54"

filename: "{wandb.run.name}-{epoch}-{val_loss:.2f}" --> "swept_sky_54-epoch10-loss0.54"

I am not sure if this is already possible but I think the filename function in the ModelCallback can only access metrics so far. 
 # Suggested Solution: 
 ### Possible Solutions 

The existing features of PyTorch Lightning may not have direct functionality to include the wandb run name in the checkpoint file name, but this can be achieved with certain workarounds. Here are a few solutions:

#### Solution 1: Using Checkpoint Callback with formatted strings
One approach is to use the `ModelCheckpoint` callback of PyTorch lightning to save checkpoints with customized names. We can leverage python's formatted string literals to include any variable into the checkpoint file name.

Here is an example of how to create a formatted string for the checkpoint name, and use it with `ModelCheckpoint`:

```python
# Assuming 'wandb_run_name' contains the name of the wandb run.
wandb_run_name = wandb.run.name
checkpoint_callback = ModelCheckpoint(
    dirpath="/your/path",
    filename=f"{wandb_run_name}"+'-{epoch:02d}-{val_loss:.2f}'
)
trainer = Trainer(callbacks=[checkpoint_callback])
trainer.fit(model)
```

The `{}` placeholders in the filename string are replaced by epoch and validation loss values automatically by PyTorch Lightning. You can add the wandb run name, or any other string to it.

#### Solution 2: Creating Custom Callback
If the file naming mechanism requires more control than what's provided by the `ModelCheckpoint`, a custom callback can be created:

```python
class CustomModelCheckpoint(Callback):

    def __init__(self, wandb_run_name):
        self.wandb_run_name = wandb_run_name

    def on_validation_end(self, trainer, pl_module):
        loss = trainer.callback_metrics['val_loss']
        epoch = trainer.current_epoch
        torch.save(
            pl_module.state_dict(), 
            f"/your/path/{self.wandb_run_name}-epoch{epoch}-loss{loss:.2f}.ckpt"
        )

wandb_run_name = wandb.run.name
trainer = Trainer(callbacks=[CustomModelCheckpoint(wandb_run_name)])
trainer.fit(model)
```
This example saves a checkpoint at the end of every validation epoch. You can modify the events when the checkpoint is saved according to your requirements.

Lastly, remember to replace "/your/path" with the directory where you want to save your model checkpoints. Remember that accessing wandb.run.name requires an active wandb run. Therefore, initialize wandb before using these solutions.

Please let us know if you have any other questions or face issues implementing these solutions.
